integration/singularity/reader.go

```diff
@@ -0,0 +1,76 @@
+package singularity
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	singularityclient "github.com/data-preservation-programs/singularity/client/swagger/http"
+	"github.com/data-preservation-programs/singularity/client/swagger/http/file"
+	"github.com/gotidy/ptr"
+)
+
+// io.ReadSeekCloser implementation that reads from remote singularity
+type SingularityReader struct {
+	client *singularityclient.SingularityAPI
+	fileID uint64
+	offset int64
+	size   int64
+}
+
+func (r *SingularityReader) Read(p []byte) (int, error) {
+	logger.Infof("buffer size: %v", len(p))
+
+	buf := bytes.NewBuffer(p)
+	buf.Reset()
```

Comment:
I am curious why buffer is being reset? it's a fresh buffer right?

Reply:
the buffer is constructed from the byte array passed into the function, but creating a new buffer like that starts the cursor at the end of the byte array, so buf.Reset() puts the cursor back at the start so the array will be overwritten instead of extended from the end